v4/weaviate/auth/auth.go

```diff
@@ -46,22 +47,22 @@ func (ab authBase) getIdAndTokenEndpoint(ctx context.Context, con *connection.Co
 				"miss-configured or you have a proxy inbetween the client and weaviate. You can test this by visiting %v.",
 				oidcConfigURL)
 
-			return "", "", nil
+			return "", []string{}, "", nil
 		}
 	default:
-		return "", "", fmt.Errorf("OIDC configuration url %s returned status code %v", oidcConfigURL, rest.StatusCode)
+		return "", []string{}, "", fmt.Errorf("OIDC configuration url %s returned status code %v", oidcConfigURL, rest.StatusCode)
 	}
 
 	endpoints, err := con.RunRESTExternal(context.TODO(), cfg.Href, http.MethodGet, nil)
 	if err != nil {
-		return "", "", err
+		return "", []string{}, "", err
 	}
 	var resultEndpoints map[string]interface{}
 	decodeEndpointErr := endpoints.DecodeBodyIntoTarget(&resultEndpoints)
 	if decodeEndpointErr != nil {
-		return "", "", err
+		return "", []string{}, "", err
 	}
-	return cfg.ClientID, resultEndpoints["token_endpoint"].(string), nil
+	return cfg.ClientID, cfg.Scopes, resultEndpoints["token_endpoint"].(string), nil
```

Comment:
Maybe want to do a type assertion check on `resultEndpoints["token_endpoint"]`, otherwise if this panics it may cause a bad UX

Reply:
done